### PR TITLE
[style] simplify build paths

### PR DESCRIPTION
**Pull request by bukepo**
This commit simplifies the build paths:

* remove cmake prefix from `script/test`
* use fixed name for simulation and posix platforms just like others

Depends-On: openthread/ot-ns#112

### CODE REVIEW

**Comment by bukepo** [comment]
this commit mutual depends on https://github.com/openthread/ot-ns/pull/112

**Comment by simonlingoogle** [comment]
> this commit mutual depends on [openthread/ot-ns#112](https://github.com/openthread/ot-ns/pull/112)

I think we should use a OTNS commit that allows this PR to pass all checks. Then, we can update OTNS to use openthread master to pass all OTNS CI checks. It's too risky to merge two PRs without passing checks. 

TIP: set OTNS_COMMIT to the latest commit of https://github.com/openthread/ot-ns/pull/112

**Comment by simonlingoogle** [comment]
I think there are many documentation that are affected by this PR, e.g. https://openthread.io/codelabs/openthread-simulation-posix#3. How do we take care of them? @bukepo @Vyrastas @jwhui

File: .github/workflows/simulation.yml [hunk 1]
```diff
@@ -89,7 +89,7 @@ jobs:
         ./bootstrap
     - name: Run OTNS Tests
       env:
-        OTNS_COMMIT: "632503d7ccbadcbdc49c6245eafc2425c80bbf04" # Sep 29, 2020
+        OTNS_COMMIT: "2d57478d9e748d09cad65cf5374942e9d96719ed" # just for CI
```
Comment (simonlingoogle) [inline .github/workflows/simulation.yml:92]: nit: provide OTNS PR number?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 
with a nit

File: .github/workflows/simulation.yml (hunk 1 above)
Comment (bukepo) [reply .github/workflows/simulation.yml:92]:
```suggestion
        OTNS_COMMIT: "2d57478d9e748d09cad65cf5374942e9d96719ed" # openthread#ot-ns#112
```

File: src/posix/README.md [hunk 2]
```diff
@@ -89,16 +89,16 @@ python cc2538-bsl/cc2538-bsl.py -b 460800 -e -w -v -p /dev/ttyUSB0 ot-rcp.bin
 ### With Simulation
 
 ```sh
-sudo wpantund -s 'system:./output/posix/x86_64-unknown-linux-gnu/bin/ot-ncp spinel+hdlc+forkpty://output/x86_64-unknown-linux-gnu/bin/ot-rcp?forkpty-arg=1'
+sudo wpantund -s 'system:./output/posix/bin/ot-ncp spinel+hdlc+forkpty://output/x86_64-unknown-linux-gnu/bin/ot-rcp?forkpty-arg=1'
```
Comment (bukepo) [inline src/posix/README.md:92]:
```suggestion
sudo wpantund -s 'system:./output/posix/bin/ot-ncp spinel+hdlc+forkpty://output/simulation/bin/ot-rcp?forkpty-arg=1'
```

File: src/posix/README.md [hunk 3]
```diff
@@ -109,9 +109,9 @@ OpenThread Posix Daemon mode uses a unix socket as input and output, so that Ope
 # build daemon mode core stack for POSIX
 make -f src/posix/Makefile-posix DAEMON=1
 # Daemon with simulation
-./output/posix/x86_64-unknown-linux-gnu/bin/ot-daemon 'spinel+hdlc+forkpty://output/x86_64-unknown-linux-gnu/bin/ot-rcp?forkpty-arg=1'
+./output/posix/bin/ot-daemon 'spinel+hdlc+forkpty://output/x86_64-unknown-linux-gnu/bin/ot-rcp?forkpty-arg=1'
```
Comment (bukepo) [inline src/posix/README.md:112]:
```suggestion
./output/posix/bin/ot-daemon 'spinel+hdlc+forkpty://output/simulation/bin/ot-rcp?forkpty-arg=1'
```

**Review by librasungirl — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**